src/backend/drm/exporter/gbm.rs

```diff
@@ -95,7 +97,7 @@ impl<A: AsFd + 'static> ExportFramebuffer<GbmBuffer> for GbmFramebufferExporter<
             #[cfg(not(all(feature = "backend_egl", feature = "use_system_lib")))]
             ExportBuffer::Wayland(buffer) => crate::wayland::dmabuf::get_dmabuf(buffer)
                 .ok()
-                .is_some_and(|buf| buf.node().is_some_and(|node| Some(node) == self.import_node)),
+                .is_some_and(|buf| buf.node().is_some_and(|node| self.import_node == node)),
```

Comment:
I'm not fully aware of the context in which this method is used, but for implementing this patch I've assumed that not having any node will never allow adding a framebuffer, even if `NodeFilter::All` is used.